packages/paraswap/src/Paraswap.ts

```diff
@@ -22,96 +23,66 @@ import {
   OPTIMISM_SEPSP2_ADDRESS,
 } from './contract-addresses.js'
 const fetcher = constructAxiosFetcher(axios) // alternatively constructFetchFetcher
-// If you're implementing swap or mint, simply duplicate this function and change the name
+const ETH_ADDRESS = '0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE'
+
 export const swap = async (swap: SwapActionParams) => {
-  // This is the information we'll use to compose the Transaction object
-  const {
-    chainId,
-    contractAddress,
-    tokenIn,
-    tokenOut,
-    amountIn,
-    amountOut,
-    recipient,
-  } = swap
+  const { chainId, contractAddress, tokenIn, tokenOut, amountIn, amountOut } =
+    swap
   const { getAugustusSwapper } = constructGetSpender({ chainId, fetcher })
   const to = contractAddress || (await getAugustusSwapper())
-  // We always want to return a compressed JSON object which we'll transform into a TransactionFilter
+  const ethUsedIn = tokenIn === Tokens.ETH
```

Comment:
Is there ever a case where if `tokenIn === Tokens.ETH` we don't use `ETH_ADDRESS`? Cause you could just do like...

```
const tokenInUsed = tokenIn === Tokens.ETH : ETH_ADDRESS ? tokenIn
```

And just always use that value to avoid needed the extra logic in each ABI param section.

In cases like ` assets: buildPathQuery(tokenIn, tokenOut),` on line 84 we could just still use the `tokenIn` (assuming that isn't supposed to use the `ETH_ADDRESS`

Reply:
I have made the recommended adjustments. Please take a quick look when you get a chance. 🙏